src/varity/hgvs_to_vcf/protein.clj

```diff
@@ -77,33 +89,22 @@
           (when (= mut-ref-aa ref-aa)
             (let [palt (mut/->short-amino-acid (:alt mut*))
                   codon-cands (codon/amino-acid->codons palt)
-                  pos-cands (cond-> pos-cands reverse? reverse)]
+                  pos-cands* (cond-> pos-cands reverse? reverse)]
               (->> codon-cands
                    (keep
                     (fn [codon*]
-                      (->> pos-cands
-                           (map-indexed
-                            (fn [idx pos]
-                              (if (one-character-substituted?
-                                   ref-codon codon* idx)
-                                (let [ref (str (nth ref-codon idx))
-                                      alt (str (nth codon* idx))]
-                                  {:vcf {:chr chr
-                                         :pos pos
-                                         :ref (cond-> ref
-                                                reverse? util-seq/revcomp)
-                                         :alt (cond-> alt
-                                                reverse? util-seq/revcomp)}
-                                   :cdna (hgvs/hgvs
-                                          (:name rg)
-                                          :cdna
-                                          (mut/dna-substitution
-                                           (rg/cds-coord pos rg)
-                                           ref
-                                           (if (= ref alt) "=" ">")
-                                           alt))}))))
-                           (remove nil?))))
-                   (flatten)))))))
+                      (when-let [[ps ref alt] (codon->variant
+                                               pos-cands* ref-codon codon*)]
+                        (let [cds-pos (rg/cds-coord (first ps) rg)
+                              cds-end (rg/cds-coord (last ps) rg)
+                              mut (if-not (= cds-pos cds-end)
+                                    (mut/dna-indel cds-pos cds-end nil alt)
```

Comment:
why don't you pass `ref` to the 3rd arg of `mut/dna-indel`? it will be used when formatting the indel with `:show-bases? true` option.